src/components/polygons/boundary-drawer.vue

```diff
@@ -0,0 +1,267 @@
+    private created() {
+        this.onModeChanged(this.mode);
+    }
+
+    @Watch('mode')
+    private onModeChanged(newMode: DrawingMode) {
+        if (this.mode === 'polygon') {
+            this.internalMode = 'before-polygon';
+        } else if (this.mode === 'box') {
+            this.internalMode = 'before-corner1';
+        }
+        this.pointerCancel();
+    }
+
+    private get svgClass() {
+        if (this.internalMode === 'before-polygon' || this.internalMode === 'polygon') {
+            return ['draw-boundary'];
+        }
+        if (this.internalMode === 'before-corner1') {
+            return ['draw-first-corner'];
+        }
+
+        if (this.internalMode === 'before-corner2') {
+            return ['draw-second-corner'];
+        }
+
+        return [];
+    }
+
+    private pointerDown($event: PointerEvent) {
+        this.activePointers.add($event.pointerId);
+        if (this.activePointers.size > 1) {
+            this.cancelOperation();
+            return;
+        }
+
+        const pt = this.eventToPoint($event);
+        this.closedPolygon = false;
+        if (this.internalMode === 'before-corner1') {
+            this.internalMode = 'before-corner2';
+        } else if (this.internalMode === 'before-polygon') {
+            this.corner1 = this.corner2 = pt;
+            this.internalMode = 'polygon';
+        }
+
+        this.corner1 = {x: pt.x, y: pt.y};
+        this.corner2 = {x: pt.x, y: pt.y};
+        this.polygonPoints = [pt];
+    }
+
+    private pointerMove($event: PointerEvent) {
+        if (this.activePointers.size > 1) {
+            return;
+        }
+
+        const pt = this.eventToPoint($event);
+        if (this.internalMode === 'before-corner2') {
+            this.corner2 = pt;
+            this.polygonPoints = [this.corner1!,
+                { x: this.corner1!.x, y: this.corner2!.y },
+                this.corner2!,
+                { x: this.corner2!.x, y: this.corner1!.y },
+            ];
+            this.closedPolygon = true;
+        } else if (this.internalMode === 'polygon') {
+            this.polygonPoints!.push(this.eventToPoint($event));
+
+            // Update the polygon's bounding box
+            if (pt.x < this.corner1!.x) {
+                this.corner1!.x = pt.x;
+            }
+            if (pt.y < this.corner1!.y) {
+                this.corner1!.y = pt.y;
+            }
+            if (pt.x > this.corner2!.x) {
+                this.corner2!.x = pt.x;
+            }
+            if (pt.y > this.corner2!.y) {
+                this.corner2!.y = pt.y;
+            }
+
+            this.closedPolygon = this.checkPolygonCloseness();
+        }
+    }
+
+    // Check if a polygon is closed.
+    // This is done by looking at the distance between the last point and the first point. If they
+    // are close enough (less than 5% of the diagonal of the polygon's bounding box),
+    // the polygon is considered closed.
+    private checkPolygonCloseness(): boolean {
```

Comment:
After using this a bit, I like that the polygon doesn't display as closed until I get close to the ending point. I think I would also like for the polygon to autoclose on mouseup no matter how far the endpoint is from the startpoint.